lib/semian/memcached.rb

```diff
@@ -0,0 +1,67 @@
+require 'semian/adapter'
+require 'memcached'
+
+module Semian
+  module Memcached
+    include Semian::Adapter
+
+    class SemianError < ::Memcached::Error
+      def initialize(semian_identifier, *args)
+        super(*args)
+        @semian_identifier = semian_identifier
+      end
+    end
+
+    ResourceBusyError = Class.new(SemianError)
+
+    def semian_identifier
+      name = semian_options && semian_options[:name]
+      ["memcached", name].compact.join("_").to_sym
+    end
+
+    def raw_semian_options
+      # disable semian when host ejection is enabled.
+      return false if !!@options[:auto_eject_hosts]
+      @options[:semian].merge(circuit_breaker: false)
```

Comment:
I guess this is okay because we'll still be protected by the bulkheads. Is this implicitly relying on the internal memcached circuit breakers to disable bad hosts?

Reply:
Correct 👍